docs/download-upload-clone/storage/brief.md

```diff
@@ -0,0 +1,43 @@
+# Storage for MarxanCloud project archives
+
+This document outlines the key points of the storage strategy for MarxanCloud
+project archives: these are the downloadable artifacts that contain all the
+relevant data and metadata for a MarxanCloud project and its scenarios, which
+can subsequently be uploaded to a MarxanCloud instance in order to set up a
+copy of the downloaded project.
+
+## Project and scenario export and import flow - storage
+
+The preparation of a MarxanCloud project archive (see main [brief
+document](../brief.md)) is articulated through the preparation of several
+"pieces", each storing a coherent part of the data and metadata that make up a
+project or scenario.
+
+Whereas some metadata pieces may take up only a few hundred bytes, core project
+geo data may result in large pieces (in the GiB order of magnitude). Handling
+such files efficiently is therefore a key requirement during preparation,
+storage, download, upload and processing while creating a MarxanCloud project
+from an uploaded archive.
+
+In broad strokes, we need to account for the following steps. Details are
+outlined in the [high-level design document](./high-level-design.md).
+
+- storing one or more archive pieces as they are prepared and finalized
+
+- preparing a final archive file that includes all the finalized pieces
+
+- storing the final archive
+
+- removing individual pieces
+
+- allowing users to download the archive for a given project
+
+- removing the archive and any related files once its validity expires
```

Comment:
I assume this would happen "long" after the export file generation ended. At this stage, shouldn't there be just one file - the final, zipped file?

Reply:
I left this open as I am not sure about this - I'd be actually open (maybe even keen?) to *not* archive all the pieces into a single, potentially humongous, zip file. The obvious benefit of a single zip is the user experience for non-tech users - get one file, handle one file...

On the other hand, especially in the case of very large zip files, we incur the extra operational "cost" (more pain than cost) of extracting our pieces from the zip file. Yes, we can uncompress and save individual files as we stream, but if anything goes wrong during this operation for a very large zip file, it's a pain. And then we hold locally, temporarily, the extracted pieces, which we could otherwise stream directly, already uncompressed/unpacked, from blob storage...

So yes, my preference so far would strongly lean on leaving the pieces in blob storage, and maybe even not bothering creating a single zip file for the end user.